Imagr/MainController.py

```diff
@@ -1105,17 +1120,153 @@ def Clone(self, source, target, erase=True, verify=True, show_activity=True):
+                                  stderr=subprocess.PIPE)
+        memsizetuple = sysctl.communicate()
+        # sysctl returns crappy things from stdout.
+        # Ex: ('hw.memsize: 1111111\n', '')
+        memsize = int(
+            memsizetuple[0].split('\n')[0].replace('hw.memsize: ', ''))
+        NSLog(u"Total Memory is %@", str(memsize))
+        # Assume netinstall uses at least 650MB of RAM. If we don't require
+        # enough RAM, gurl will timeout or cause RecoveryOS to crash.
+        availablemem = memsize - 681574400
+        NSLog(u"Available Memory for DMG is %@", str(availablemem))
+        if imaging is True:
+            filesize = Utils.getDMGSize(source)[0]
+        else:
+            filesize = Utils.getDMGSize(source.get('url'))[0]
+        NSLog(u"Required Memory for DMG is %@", str(filesize))
+        # Formatting RAM Disk requires around 5% of the total amount of
+        # bytes. Add 6% to compensate for the padding we will need.
+        paddedfilesize = int(filesize) * 1.06
+        NSLog(u"Padded Memory for DMG is %@", str(paddedfilesize))
+        if filesize is False:
+            NSLog(u"Error when calculating source size. Using original method "
+                  "instead of gurl...")
+            return False, True
+        elif imaging is True and 9000000000 > memsize:
+            NSLog(u"Feature requires more than 9GB of RAM. Using asr "
+                  "instead of gurl...")
+            return False, True
+        elif int(paddedfilesize) > availablemem:
+            NSLog(u"Available Memory is not sufficient for source size. "
+                  "Using original method instead of gurl...")
+            return False, True
+        elif 8000000000 > memsize:
+            NSLog(u"Feature requires at least 8GB of RAM. Using original "
+                  "method instead of gurl...")
+            return False, True
+        else:
+            sectors = int(paddedfilesize) / 512
+            ramstring = "ram://%s" % str(sectors)
+            NSLog(u"Amount of Sectors for RAM Disk is %@", str(sectors))
+            ramattachcommand = ["/usr/bin/hdiutil", "attach", "-nomount",
+                                ramstring]
+            ramattach = subprocess.Popen(ramattachcommand,
+                                         stdout=subprocess.PIPE,
+                                         stderr=subprocess.PIPE)
+            devdisk = ramattach.communicate()
+            # hdiutil returns some really crappy things from stdout
+            # Ex: ('/dev/disk20     \t         \t\n', '')
+            devdiskstr = devdisk[0].split(' ')[0]
+            randomnum = random.randint(1000000, 10000000)
+            ramdiskvolname = "ramdisk" + str(randomnum)
+            NSLog(u"RAM Disk mountpoint is %@", str(ramdiskvolname))
+            # Only check for apfs_image status if imaging, startosinstall
+            # should just use apfs since it's meant for 10.13 and higher.
```

Comment:
`startosinstall` doesn't format a disk image at all; all the disk images containing "Install macOS High Sierra.app" I've used are HFS+ formatted, which would be important if you were trying to upgrade a Mac running macOS < 10.12.x.  I'm not convinced of the need to format your ramdisk as anything other than HFS+.

Reply:
The only reason I am choosing this is because APFS = future and HFS could potentially go away in a future NBI. I'd rather not have to fix this later if that's the case.

Reply:
Pretty sure every single ramdisk that gets mounted in the macOS Installer environment (even 10.13) is formatted HFS+. I don't think HFS+ support will go away from macOS for a very long time. But your call.

Reply:
In your opinion, should `startosinstall` (in an Imagr content, not in the fact it can work) support 10.11 10.12 NBIs? If so, then I need probably need to do HFS.